src/processors/models/processorsManager.js

```diff
@@ -28,59 +27,71 @@ module.exports.createProcessor = async function (processor) {
 };
 
 module.exports.getAllProcessors = async function (from, limit) {
-    return databaseConnector.getAllProcessors(from, limit);
+    let allProcessors = await databaseConnector.getAllProcessors(from, limit);
+    return allProcessors;
 };
 
 module.exports.getProcessor = async function (processorId) {
     const processor = await databaseConnector.getProcessorById(processorId);
     if (processor) {
         return processor;
     } else {
-        const error = generateProcessorNotFoundError();
+        const error = generateError(ERROR_MESSAGES.NOT_FOUND, 404);
         throw error;
     }
 };
 
 module.exports.deleteProcessor = async function (processorId) {
     const tests = await testsManager.getTestsByProcessorId(processorId);
     if (tests.length > 0) {
-        throw generateProcessorIsUsedByTestsError(tests.map(test => test.name));
+        let testNames = tests.map(test => test.name);
+        let message = `${ERROR_MESSAGES.PROCESSOR_DELETION_FORBIDDEN}: ${testNames.join(', ')}`;
+        throw generateError(message, 409);
     }
     return databaseConnector.deleteProcessor(processorId);
 };
 
 module.exports.updateProcessor = async function (processorId, processor) {
     const oldProcessor = await databaseConnector.getProcessorById(processorId);
     if (!oldProcessor) {
-        throw generateProcessorNotFoundError();
+        throw generateError(ERROR_MESSAGES.NOT_FOUND, 404);
     }
     if (oldProcessor.name !== processor.name) {
         const processorWithUpdatedName = await databaseConnector.getProcessorByName(processor.name);
         if (processorWithUpdatedName) {
-            throw generateProcessorNameAlreadyExistsError();
+            throw generateError(ERROR_MESSAGES.PROCESSOR_NAME_ALREADY_EXIST, 400);
         }
     }
 
     processor.created_at = oldProcessor.created_at;
-    fileManager.validateJavascriptContent(processor.javascript);
+    let exportedFunctions = verifyJSAndGetExportedFunctions(processor.javascript);
+    processor.exported_functions = exportedFunctions;
     await databaseConnector.updateProcessor(processorId, processor);
     return processor;
 };
 
-function generateProcessorNotFoundError() {
-    const error = new Error(ERROR_MESSAGES.NOT_FOUND);
-    error.statusCode = 404;
-    return error;
-}
+function verifyJSAndGetExportedFunctions(src) {
+    let exportedFunctions;
+    try {
+        let m = new module.constructor();
+        m.paths = module.paths;
+        m._compile(src, 'none');
```

Comment:
from `_compile` documentation (https://github.com/nodejs/node-v0.x-archive/blob/master/lib/module.js#L380) the `'none'` provided to this function is the filename which seems irrelevant so erase that parameter